File: hydrograph.engine/hydrograph.engine.spark/src/main/scala/hydrograph/engine/spark/components/adapter/InputMysqlAdapter.scala
package hydrograph.engine.spark.components.adapter

import hydrograph.engine.core.component.generator.InputMysqlEntityGenerator
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent
import hydrograph.engine.spark.components.InputMysqlComponent
import hydrograph.engine.spark.components.adapter.base.InputAdatperBase
import hydrograph.engine.spark.components.base.InputComponentBase
import hydrograph.engine.spark.components.platform.BaseComponentParams

/**
  * Created by santlalg on 12/7/2016.
  */
class InputMysqlAdapter(typeBaseComponent: TypeBaseComponent) extends InputAdatperBase{
  private var inputMysql:InputMysqlEntityGenerator=null
  private var sparkIMysqlComponent:InputMysqlComponent=null

  override def createGenerator(): Unit = {
    inputMysql=new InputMysqlEntityGenerator(typeBaseComponent)
  }

  override def createComponent(baseComponentParams: BaseComponentParams): Unit = {
    sparkIMysqlComponent= new InputMysqlComponent(inputMysql.getEntity,
      baseComponentParams)
  }

  override def getComponent(): InputComponentBase = sparkIMysqlComponent
}

File: hydrograph.engine/hydrograph.engine.cascading/src/main/java/hydrograph/engine/adapters/InputMysqlAdapter.java
package hydrograph.engine.adapters;

import hydrograph.engine.adapters.base.InputAdapterBase;
import hydrograph.engine.cascading.assembly.InputMysqlAssembly;
import hydrograph.engine.cascading.assembly.base.BaseComponent;
import hydrograph.engine.cascading.assembly.infra.ComponentParameters;
import hydrograph.engine.core.component.generator.InputMysqlEntityGenerator;
import hydrograph.engine.jaxb.commontypes.TypeBaseComponent;

public class InputMysqlAdapter extends InputAdapterBase{

	InputMysqlAssembly mysqlComponent;
	InputMysqlEntityGenerator entityGenerator;
	
	public InputMysqlAdapter(TypeBaseComponent component) {
		entityGenerator=new InputMysqlEntityGenerator(component);
		
	}
	
	
	@Override
	public BaseComponent getAssembly() {
		return mysqlComponent;
	}

	@Override
	public void createAssembly(ComponentParameters componentParameters) {
		mysqlComponent=new InputMysqlAssembly(entityGenerator.getEntity(), componentParameters);
	}

}
